static/src/javascripts/projects/commercial/modules/liveblog-adverts.ts

```diff
@@ -69,31 +76,31 @@ const getSlotName = (isMobile, slotCounter) => {
 	return `inline${slotCounter + 1}`;
 };
 
-const insertAds = (paras) => {
+const insertAds: SpacefinderWriter = async (paras) => {
 	const isMobile = getBreakpoint() === 'mobile';
-	let fastdomPromises = [];
+	const fastdomPromises = [];
 	for (let i = 0; i < paras.length && AD_COUNTER < MAX_ADS; i += 1) {
 		const para = paras[i];
-		if (para && para.parentNode) {
+		if (para.parentNode) {
 			const adSlot = createAdSlot('inline', {
 				name: getSlotName(isMobile, AD_COUNTER),
 				classes: 'liveblog-inline',
 			});
 			// insert the ad slot container into the DOM
 			const result = fastdom.mutate(() => {
-				para.parentNode.insertBefore(adSlot, para.nextSibling);
+				para.parentNode?.insertBefore(adSlot, para.nextSibling);
 			});
 			fastdomPromises.push(result);
 			// load and display the advert via GAM
 			addSlot(adSlot, false);
 			AD_COUNTER += 1;
 		}
 	}
-	return Promise.all(fastdomPromises);
+	await Promise.all(fastdomPromises);
```

Comment:
Would it be _slightly_ better to return the `Promise` here?

Whilst in this context its functionally equivalent I think it signals better intent of how it will be used?

aiui `insertAds` gets used by `SpaceFiller.fillSpace()` as a return from a `then()` which will resolve the returned promise.

Reply:
We agreed that `await` is okay for now. We are restricted by the return type of `SpacefinderWriter` so that should be tackled first.